onnxruntime/core/platform/env.h

```diff
@@ -193,11 +193,13 @@ class Env {
   // loading a library.  The rules for determining the exact location of the
   // library are platform-specific and are not documented here.
   //
+  // global_symbols only has an effect on unix, where a value of true means to load with RTLD_GLOBAL vs RTLD_LOCAL
+  // 
   // On success, returns a handle to the library in "*handle" and returns
   // OK from the function.
   // Otherwise returns nullptr in "*handle" and an error status from the
   // function.
-  virtual common::Status LoadDynamicLibrary(const std::string& library_filename, void** handle) const = 0;
+  virtual common::Status LoadDynamicLibrary(const std::string& library_filename, bool global_symbols, void** handle) const = 0;
```

Comment:
nit: not for this PR. but might want to consider making this a default param since most of the code wouldn't care about it plus what if we miss some cases where we should be passing false.

Reply:
Yeah, I thought about that, but I wanted to leave the 'out' parameter at the end. Or to have it be an actual flag with a name, as a bool isn't easy to extend or understand without the comments.

Reply:
Overloading the function is another option along with an enum to make it self documenting.